Sources/Views/Cells/AudioMessageCell.swift

```diff
@@ -47,25 +47,51 @@ open class AudioMessageCell: MessageContentCell {
         return durationLabel
     }()
 
+    private lazy var activityIndicatorView: UIActivityIndicatorView = {
+        let activityIndicatorView = UIActivityIndicatorView(style: .gray)
+        activityIndicatorView.hidesWhenStopped = true
+        activityIndicatorView.isHidden = true
+        return activityIndicatorView
+    }()
+
     public lazy var progressView: UIProgressView = {
         let progressView = UIProgressView(progressViewStyle: .default)
         progressView.progress = 0.0
         return progressView
     }()
 
+    public var isLoadingIndicatorVisible: Bool {
+        get {
+            return activityIndicatorView.isHidden
+        }
+        set {
+            if newValue {
+                activityIndicatorView.isHidden = false
+                playButton.isHidden = true
+                activityIndicatorView.startAnimating()
+            }
+            else {
+                playButton.isHidden = false
+                activityIndicatorView.stopAnimating()
+            }
+        }
+    }
+    
     // MARK: - Methods
 
     /// Responsible for setting up the constraints of the cell's subviews.
     open func setupConstraints() {
         playButton.constraint(equalTo: CGSize(width: 25, height: 25))
-        playButton.addConstraints(left: messageContainerView.leftAnchor, centerY: messageContainerView.centerYAnchor, leftConstant: 5)
+        playButton.addConstraints(left: messageContainerView.leftAnchor, centerY: messageContainerView.centerYAnchor, leftConstant: 10)
```

Comment:
What was the reason behind adjusting the left constant?

Reply:
To answer the question, probably a remnant of earlier attempts at positioning.

`UIActivityIndicatorView` is not generally a component that can be resized by changing the frame, so only certain styles of it will fit properly in this case. But in any case, users can subclass and override their layout.